### ISSUE TITLE
Page no longer editable after certain text added to web part configuration 

### ISSUE THREAD

**Issue by TjWheeler** [SharePoint/sp-dev-docs#4774]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
Regardless of Web Part configuration, when clicking the Edit button on a page in a modern site, the page should go into Edit mode.

#### Observed Behavior
When a web part is saved with specific text (detailed in steps to reproduce) in the configuration properties, and clicking the Edit button on the page, the following is observed:

A dialog is displayed with:
`You cannot edit this page`
`We're sorry, we encountered an unexpected error. Please try again, if the problem persists please contact your Administrator.`

A POST call is made when clicking edit.
`POST /sites/ModernSite/_api/sitepages/pages(69)/checkoutpage`
The response received has a 500 status code, and a body of:
```
{"odata.error":{"code":"-1, Newtonsoft.Json.JsonReaderException","message":{"lang":"en-US","value":"Unterminated string. Expected delimiter: \". Path 'properties.description', line 1, position 315."}}}
```

#### Steps to Reproduce

Create a Hello World Web Part using all default values:
1. yo @microsoft/sharepoint
1. Bundle and Package using the --ship option
1. Deploy to App Catalog and a Modern Team site
1. Add the new web Part to a page and set the Description property to:
```<h1>[[Test]]</h1>```
1. Click Save as draft
1. Click Edit

*Note: This isn't something I could reproduce in the workbench.
#### Other details
Framework version 1.9.1 with No JavaScript framework.
includeClientSideAssets is true.

**Comment by johnguy0** [comment]
@TjWheeler The `Description` property is supposed to be a text only field. If you are persisting HTML data, the `propertiesMetadata` property on the web part should reflect this by following this [documentation](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/guidance/integrate-web-part-properties-with-sharepoint#specify-web-part-property-value-type). Can you try setting `isHtmlString` to `true` and let us know if you are still seeing this issue?

**Comment by TjWheeler** [comment]
Hi @johnguy0, thanks for pointing out that documentation.  I hadn't tried that.

Setting it to isHtmlString: true did indeed fix the Page Corruption issue.  The page can now be saved and edited without issue.

Unfortunately I can't really use that for my scenario, because the data is manipulated within the web part. 

After setting the `description` field to an html string and saving the value `[[test]]` into that field in the configuration of the web part on a page, the data is changed to 
```
<a id="0::test|test" class="ms-missinglink" href="/sites/mydevsite/SitePages/test.aspx">test</a>
```

So it seems that SharePoint is interpreting `[[test]]` as a link and converting it.  
Here is the update I made based on your info:
```typescript
protected get propertiesMetadata(): IWebPartPropertiesMetadata {
    return {
      title: { isSearchablePlainText: true },
      description: { isHtmlString: true }
    };
  }
```

So to summarize, the page no longer being editable is fixed by the `propertiesMetadata` method and declaring the value as an html string, however since SharePoint manipulates that data, I cannot use it in my web part.  

I have changed my solution to use `((` and `))` and that works ok.  Thanks for your help.

**Comment by johnguy0** [comment]
Yes, unfortunately the double brackets are a part of the legacy [wiki link syntax](https://support.office.com/article/create-and-edit-a-wiki-dc64f9c2-d1a2-44b5-ac59-b9d535551a32).

Another workaround could be to encode/decode the brackets if you didn't want to change your original syntax.
